src/orders/service/orders.service.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  BadRequestException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
+import OrdersRepository from '../repository/orders.repository';
+import Order from '../domain/order.domain';
+import { OrderDetailsService } from './orderDetails.service';
+import OrderDetail from '../domain/orderDetail.domain';
+
+@Injectable()
+export class OrdersService {
+  constructor(
+    private ordersRepository: OrdersRepository,
+    private orderDetailsService: OrderDetailsService,
+  ) {}
+
+  getOrders(): Promise<Order[]> {
+    return this.ordersRepository.findAll();
+  }
+
+  async getOrderById(id: string): Promise<Order | null> {
+    const order = await this.ordersRepository.findOne(id);
+    if (!order) {
+      throw new NotFoundException("The order can't be found");
+    }
+    return order;
+  }
+
+  async createOrder(order: Order, orderDetails: OrderDetail[]): Promise<Order> {
+    await this.validateOrder(orderDetails);
+
+    const savedOrder = await this.ordersRepository.save(order);
+    await this.orderDetailsService.decreaseStock(orderDetails);
+    orderDetails.map((orderDetail) => {
+      orderDetail.orderId = savedOrder.id;
+    });
+    await this.orderDetailsService.saveOrderDetails(orderDetails);
```

Comment:
You don't need to modify anything now, but keep in mind the following issue for the future:
- in this method there are multiple save/update operations
- some of them might succeed, some of them might fail (from various reasons, e.g, database connection issues)
- we never want to have partial writes because we will have inconsistent data (e.g., don't save the order without order details; rollback the order) 

So, the solution is using **transactions**:
- https://docs.nestjs.com/techniques/database#typeorm-transactions
- https://www.npmjs.com/package/typeorm-transactional (the Decorator simplifies implementation)

Reply:
We've also used transactions at university, and you're right, this is the right way to do it, I don't know how I didn't think about this. Thank you, I'll keep it in mind and I'll have enough time after the chapters I'm gonna modify it for sure!